File: src/go/thread-2.go
package main

import (
    "fmt"
    "time"
)

func saudar(nome string, vezes int) {
    for i := 0; i < vezes; i++ {
            fmt.Printf("Olá, %s! (mensagem %d)\n", nome, i+1)
    }
}

func main() {
go saudar("Maria", 3)

time.Sleep(1 * time.Second)

}

File: src/python/thread-2.py
# thread - passar valores por parâmetro para thread
# 

# importar as libs
import threading

# função que será executada na thread
def saudar(nome, vezes):
    for i in range(vezes):
        print(f"Olá, {nome}! (mensagem {i+1})")

def main():
    # Criar thread com argumentos
    thread = threading.Thread(target=saudar, args=("Maria", 3))
    thread.start()
    thread.join()

if __name__ == "__main__":
    main()
